### DISCUSSION TITLE
Newbie question on getting Pixel Format from an image

### DISCUSSION THREAD

**Discussion by iMrShadow** [SixLabors/ImageSharp#2682] [Q&A] [answered]
Hello! Is there a way to know what pixel / color format is used in the original image? 

For example, I want to know whether the original pixel format is in
Format32bppArgb
Format16bppArgb1555
Format16bppRgb555
Format16bppRgb565
Format24bppRgb
Format8bppIndexed 
and [others](https://learn.microsoft.com/en-us/dotnet/api/system.drawing.imaging.pixelformat?view=dotnet-plat-ext-8.0).
But I want their equivalent counter parts from this [table](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.PixelFormats.html#structs): 

I am working with .png, .jpg, .tiff, .bmp and .tga, and then converting them into .dds format.

**Comment by antonfirsov** [comment, marked as answer]
Note that System.Drawing's [`Image.PixelFormat`](https://learn.microsoft.com/en-us/dotnet/api/system.drawing.image.pixelformat) doesn't necessarily represent the format used in the "original image", if you mean the encoded file by that. While it's correct for a PNG, note that JPEG stores images in YCbCr color space, meaning if you don't ask for a specific format when decoding, the outcome, `new Bitmap("img.jpg").PixelFormat == PixelFormat.Format24bppRgb` is not really accurate. Long story short `Image.PixelFormat` is the format System.Drawing decides to internally represent your image in when you are not asking for a specific format at decoding, and not the absolute thruth about the source image.

ImageSharp will also pick an arbitrary pixel format when using the non-generic `Image.Load()`, but I don't think that information is useful in any way for you. There is functionality to acquire information/metadata about the original image format if that is what you are after:

```C#
// The sample is using 3.1.* API
ImageInfo info = Image.Identify("my.png");
PixelTypeInfo pixelType = info.PixelType;
Console.WriteLine($"bpp: {pixelType.BitsPerPixel}");

// There is no way to get more information from pixelType at the moment,
// however you can observe format specific information:
if (info.Metadata.TryGetPngMetadata(out PngMetadata pngMetadata))
{
    // pngMetadata has plenty of properties
}
```

> and then converting them into .dds format.

If you are resaving the image anyways, I wonder what do you need the pixel format information for?

**Reply by iMrShadow** [reply]
Thanks for the answer! The pixel format from System.Drawing was used in old code and I originally thought I needed it, but at the end I didn't. 
The idea was displaying image data (width, length, depth and others) on the screen for information.

I have another question, if I may ask. I want to know if an image is transparent/has an alpha channel. However, when I use ImageInfo and extract the PixelFormat.AlphaRepresentation, that is always null (even for images which are transparent). That seems like a strange behaviour, unless I am misunderstanding something

**Reply by antonfirsov** [reply]
Unfortunately, filling `AlphaRepresentation` (or any values other than `BitsPerPixel`) on `Image.Identify` is not yet implemented. You need to check the format-specific metadata to figure those things out.

cc @JimBobSquarePants in case you want to put this to the V4 feature list.

**Reply by JimBobSquarePants** [reply]
I’m toying with the idea of dropping it for decoding and making it representative of pixel formats only since we have the per-format metadata. I want to do some normalisation there also.

**Reply by antonfirsov** [reply]
It is harder to code against from user POV. One needs to iterate through all the arbitrary properties of all format metadatas.

**Reply by JimBobSquarePants** [reply]
Someone would need to contribute this code then. We'd have to go through each format and assign the corresponding properties. I dread to think what that is like for Tiff.
